interfaces/builtin/userns.go

```diff
@@ -0,0 +1,104 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	apparmor_sandbox "github.com/snapcore/snapd/sandbox/apparmor"
+	"github.com/snapcore/snapd/strutil"
+)
+
+const userNSSummary = `allows the ability to use user namespaces`
+
+// This interface is super-privileged
+const userNSBaseDeclarationSlots = `
+  userns:
+    allow-installation: false
+    deny-connection: true
+    deny-auto-connection: true
+`
+
+const userNSConnectedPlugAppArmorSupported = `
+# Description: allow to use user namespaces
+
+userns,
+`
+
+const userNSConnectedPlugSeccomp = `
+# allow the use of unshare to use new user namespaces
+unshare
```

Comment:
should we also allow `clone`?

Reply:
the comment could be updated. Their is unsharing via clone creating a new user namespace and then there is the unshare system call. From the hook pov they are the same and we can't differentiate, but the comment could say something along the lines of
# allow the use of new user namespaces via clone and unshare

Reply:
I think @gegarcia   is right - the LSM hook may not know the difference between clone and unshare but I assume the seccomp filter does - so I will add clone here as well.